q2cli/builtin/tools.py

```diff
@@ -80,40 +80,149 @@ def export_data(input_path, output_path, output_format):
+    matches = []
+    for word in words:
+        matches += get_close_matches(word,
+                                     possibilities,
+                                     n=len(possibilities),
+                                     cutoff=cutoff)
+        # substring search
+        if cutoff != 1:
+            for possibility in possibilities:
+                if word.lower() in possibility.lower():
+                    if possibility not in matches:
+                        matches.append(possibility)
+
+    return matches
+
+
+@tools.command(
+        name='show-types',
+        help='List the available semantic types.',
+        short_help='',
+        cls=ToolCommand
+)
+@click.argument('types', nargs=-1)
+@click.option('--strict', is_flag=True,
+              help='Show only exact matches for the type argument(s).')
+@click.option('--tsv', is_flag=True,
+              help='Print as machine readable tab separated values.')
+def show_types(types, strict, tsv):
+    pm = q2cli.util.get_plugin_manager()
+
+    if types and strict:
+        matches = get_matches(types, list(pm.artifact_classes), 1)
+    elif types:
+        matches = get_matches(types, list(pm.artifact_classes))
+    else:
+        matches = list(pm.artifact_classes)
+
+    descriptions = {}
+    for match in sorted(matches):
+        description = pm.artifact_classes[match].description
+        descriptions[match] = description
+
+    print_descriptions(descriptions, tsv)
+
+
+@tools.command(
+        name='show-formats',
+        help='List the availabe formats.',
+        short_help='',
+        cls=ToolCommand
+)
+@click.argument('formats', nargs=-1)
+@click.option('--importable', is_flag=True,
+              help='List the importable formats.')
+@click.option('--exportable', is_flag=True,
+              help='List the exportable formats.')
+@click.option('--strict', is_flag=True,
+              help='Show only exact matches for the format argument(s).')
+@click.option('--tsv', is_flag=True,
+              help='Print as machine readable tab separated values.')
+def show_formats(formats, importable, exportable, strict, tsv):
+    if importable and exportable:
+        raise click.UsageError("'--importable' and '--exportable' flags are "
+                               "mutually exclusive.")
+    if not importable and not exportable:
+        raise click.UsageError("One of '--importable' or '--exportable' flags "
+                               "is required.")
+
+    pm = q2cli.util.get_plugin_manager()
+    portable_formats = pm.importable_formats if importable \
+        else pm.exportable_formats
+    available_formats = list(portable_formats)
+
+    if formats and strict:
+        matches = get_matches(formats, available_formats, 1)
+    elif formats:
+        matches = get_matches(formats, available_formats)
+    else:
+        matches = available_formats
 
-    import q2cli.util
+    descriptions = {}
+    for match in sorted(matches):
```

Comment:
A side-effect of sorting here is that you lose the sorting that `get_close_matches` provides, which is sorting by decreasing similarity. That can lead to unusual cases like the example I provided above, where the least similar match (`ImmutableMetadata`) was presented first. 

I think you should leave sorting as-is when calling `get_matches`, and sort alphabetically like you're doing here in the case where `len(formats) == 0`.